src/js_realm.hpp

```diff
@@ -213,6 +213,68 @@ class RealmDelegate : public BindingContext {
     friend class RealmClass<T>;
 };
 
+
+template<typename T>
+class ShouldCompactOnLaunchFunctor {
```

Comment:
We could refactor this in a generic functor class that enables us to use it in anywhere we need to invoke the callback on main thread and wait for the result. This implementation is too specific and next time we need it we will need to do it again.